### PR TITLE
[ServiceBus] ClientLibraryInformation serializes the current information once and then no longer uses reflection

### PR DESCRIPTION
**Pull request by danielmarbach**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Core/ClientLibraryInformation.cs [hunk 1]
```diff
@@ -55,6 +55,11 @@ internal sealed class ClientLibraryInformation
         [Description("user-agent")]
         public string UserAgent => $"azsdk-net-{ Product }/{ Version } ({ Framework }; { Platform })";
 
+        /// <summary>
+        /// Client Information properties serialized with normalized names
+        /// </summary>
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Core/ClientLibraryInformation.cs:60]:
```suggestion
        /// <summary>
        ///   Client Information properties serialized with normalized names
        /// </summary>
        ///
```

consistency nit.  😄

**Review by jsquire — APPROVED** [review]
Thanks, @danielmarbach!

**Comment by JoshLove-msft** [comment]
Would you mind adding a test for this @danielmarbach? This change makes me a little nervous since we don't appear to have any coverage of this code.

**Comment by danielmarbach** [comment]
Sure no problem. Will do

**Comment by danielmarbach** [comment]
@JoshLove-msft done

File: sdk/servicebus/Azure.Messaging.ServiceBus/tests/Core/ClientLibraryInformationTests.cs [hunk 2]
```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/tests/Core/ClientLibraryInformationTests.cs:4]: This looks awfully familiar.    😛

Comment (danielmarbach) [reply sdk/servicebus/Azure.Messaging.ServiceBus/tests/Core/ClientLibraryInformationTests.cs:4]: Must have been written by a very smart person :p

**Comment by danielmarbach** [comment]
Here the EventHubs part too https://github.com/Azure/azure-sdk-for-net/pull/30038

**Comment by danielmarbach** [comment]
Anything required for the change log? I was a bit hesitant because it is an internal only change. I haven't measured it but since we did the reflection and materialization for every connection scope this should give some noticeable boost.

**Review by JoshLove-msft — APPROVED** [review]

**Comment by JoshLove-msft** [comment]
> Anything required for the change log? I was a bit hesitant because it is an internal only change. I haven't measured it but since we did the reflection and materialization for every connection scope this should give some noticeable boost.

This would probably go under the "Other Changes" section, but it is up to you!

**Pull request merged**